thorlcr/activities/csvread/thcsvrslave.cpp

```diff
@@ -240,25 +240,33 @@ class CCsvReadSlaveActivity : public CDiskReadSlaveActivityBase
         {
             if (sentHeaderLines->testSet(subFile))
                 return;
-            unsigned which = gotHeaderLines->scan(0, false);
-            if (which == subFiles) // all received
+
+            /* Before we can send state of headerLines of subfiles,
+             * need to have received any updates from previous worker.
+             * The previous worker will have sent updates as it progressed,
+             * and info. re. all files it is not dealing with (and all remaining if stopped) */
+            while (true)
             {
-                bool someLeft=false;
-                unsigned hL=0;
-                for (; hL<subFiles; hL++)
-                {
-                    if (headerLinesRemaining[hL])
-                    {
-                        someLeft = true;
-                        break;
-                    }
-                }
-                if (!someLeft)
+                unsigned which = gotHeaderLines->scan(0, false);
+                if (which == subFiles) // all received
+                    break;
+                getHeaderLines(which);
```

Comment:
Not in this code, but the testSet() inside getHeaderLines() shouldn't be in an assert.  On first reading I assumed it didn't modify gotHeaderLines

Reply:
verifyex rather than assertex perhaps?

Reply:
yes, not new, but should have been a verifyex, I'll change.

Reply:
Fixed.
@ghalliday @richardkchapman - please re-review